docs/contributing.rst

```diff
@@ -80,6 +80,42 @@ guidelines when you're going to report a new bug.
 Writing patches
 ===============
 
+Before you can submit a change to the Scrapy code base, you need to:
+
+#. Learn the basics of Git_ and Github_
+
+#. Fork_ the Scrapy repository
+
+#. Clone_ your fork
+
+#. Configure_ upstream repository
+
+#. Install Scrapy using::
+  
+         pip install -e .
+
+   NOTE: Consider setting up a `Virtual Environment`_ before installing Scrapy. Virtual environment is used to create an isolated environment for 
```

Comment:
I wonder if, instead of suggesting this in a note, we should make the creation of a virtual environment a step of its own before installing Scrapy.